smack-extensions/src/main/java/org/jivesoftware/smackx/muc/MultiUserChatManager.java

```diff
@@ -199,6 +206,33 @@ public void processStanza(Stanza packet) {
         };
         connection.addAsyncStanzaListener(invitationPacketListener, INVITATION_FILTER);
 
+        // Listens for all messages that include an XEP-0249 GroupChatInvitation extension and fire the invitation
+        // listeners
+        StanzaListener directInvitationStanzaListener = new StanzaListener() {
+            @Override
+            public void processStanza(Stanza stanza) {
+                final Message message = (Message) stanza;
+                GroupChatInvitation invite =
+                    stanza.getExtension(GroupChatInvitation.class);
+
+                // Fire event for invitation listeners
+                final MultiUserChat muc = getMultiUserChat(invite.getRoomAddress());
+                final XMPPConnection connection = connection();
+                final EntityJid from = message.getFrom().asEntityJidIfPossible();
+                if (from == null) {
+                    LOGGER.warning("Group Chat Invitation from non entity JID: '" + message.toXML() + "'");
```

Comment:
Do not spit out the whole XML of the causing stanza if it is not a debug message.
```suggestion
                    LOGGER.warning("Group Chat Invitation from non entity JID in '" + message + "'");
```